### PR TITLE
feature(frontend): render map on page using leaflet and openstreetmap

### LINKED ISSUE #59: 🚀 Feature: render map on page using Leaflet and OpenStreetMap
Add map rendering compabilities in backend. basic scaffolds in frontend, there set to convert to react once map is live.

### PR DESCRIPTION
**Pull request by jaysabe**
### *What* change does this PR introduce?

Added map render to main page.
### *Why* was this change needed?
Core feature for soft launch of program.

Closes #59 
### Other information (screenshots)

succ_map_render

### Checklist

- [x] 📝 I have reviewed my code prior to submission
- [x] ❔ Issue number has been linked (if applicable)
- [x] 🧪 Tests have been added or updated (if applicable)
- [ ] 📖 Documentation has been updated (if applicable)
- [x] 📸 Screenshots added (if applicable)

### CODE REVIEW

File: packages/frontend/package.json [hunk 1]
```diff
@@ -30,17 +30,27 @@
     "@reduxjs/toolkit": "^2.2.1",
```
Comment (CeeJayyy007) [inline packages/frontend/package.json:30]: **issue (blocking):** Please, review this file and ensure that you installed the right packages and in their appropriate sections, for example, `commitlint` is installed in the root `package.json` and doesn't belong here. Also, we have the dependencies and devDependencies sections. Dependencies are for packages that are used in development and production while devDependencies are packages used only during development and I believe the leaflet packages are required for proper functioning of the app in production. Therefore, kindly review and ensure you installed packages correctly and in their appropriate sections. Thank you

File: package.json [hunk 2]
```diff
@@ -29,10 +29,18 @@
   "devDependencies": {
     "@commitlint/cli": "^19.0.3",
     "@commitlint/config-conventional": "^19.0.3",
+    "@types/leaflet": "^1.9.9",
```
Comment (CeeJayyy007) [inline package.json]: **issue (blocking):** The leaflet types, react , react-dom packages etc shouldn't be installed here. They should be installed in the frontend package. Kindly review and remove any unrequired package

File: packages/frontend/src/components/map/Map.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+function Map() {
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      require('leaflet');
+    }
+  }, []);
+  return (
+    <MapContainer
+      data-testid="map"
+      style={{
```
Comment (CeeJayyy007) [inline packages/frontend/src/components/map/Map.tsx]: **nitpick (non-blocking):** Please review the styling convention as we are currently using `TailwindCSS` classNames. This is non-blocking but would be good to stick to stick to a unified system

File: packages/frontend/__tests__/Home.test.tsx [hunk 4]
```diff
@@ -2,8 +2,20 @@ import { render, screen } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import Home from '../src/app/page';
 
+jest.mock('react-leaflet', () => ({
```
Comment (CeeJayyy007) [inline packages/frontend/__tests__/Home.test.tsx:5]: **praise:** Great work with the test!

File: packages/frontend/src/components/map/Map.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+'use client';
+import React, { useEffect } from 'react';
+import { MapContainer, Marker, TileLayer, Popup } from 'react-leaflet';
+// import L from 'leaflet';
```
Comment (CeeJayyy007) [inline packages/frontend/src/components/map/Map.tsx]: **nitpick (non-blocking):** Remove commented part of code from PR.  This should have been caught during linting but I guess our linting is not strict

File: packages/frontend/src/components/map/Map.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import 'leaflet/dist/leaflet.css';
+
+// const position: [number, number] = [51.505, -0.09];
+
+function Map() {
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      require('leaflet');
```
Comment (CeeJayyy007) [inline packages/frontend/src/components/map/Map.tsx]: **nitpick (non-blocking):** use import instead here `import('leaflet')` to stick to modern ES6 convention.

File: packages/frontend/src/components/map/Map.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      }}
+      center={[51.505, -0.09]}
+      zoom={13}
+      scrollWheelZoom={false}
+    >
+      <TileLayer
+        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+      />
+      <Marker position={[51.505, -0.09]}>
```
Comment (CeeJayyy007) [inline packages/frontend/src/components/map/Map.tsx]: **nitpick (non-blocking):** We could extract this into a variable and place it at the top of the file for easy readability considering that it is hard-coded

File: packages/frontend/package.json (hunk 1 above)
Comment (jaysabe) [reply packages/frontend/package.json:30]: are you saying that the leaflet stuff needs to be in both the Dependencies and devDependencies or just Dependencies?

Comment (CeeJayyy007) [reply packages/frontend/package.json:30]: The leaflet stuff needs to be in only the dependencies. Also, we are using a monorepo (lerna) and so dependencies shared by packages (frontend and backend) like `commitlint` should only be in the root package.json and nowhere else. Please refer to older version of the app to get a clearer picture of this. Thank you

File: package.json [hunk 8]
```diff
@@ -27,12 +27,15 @@
...
-    "@commitlint/config-conventional": "^19.0.3",
+    "@commitlint/cli": "^19.3.0",
+    "@commitlint/config-conventional": "^19.2.2",
     "husky": "^9.0.11",
     "lerna": "^8.1.2"
   },
   "workspaces": [
     "./packages/*"
-  ]
+  ],
```
Comment (CeeJayyy007) [inline package.json:37]: **question (blocking):** Please, is this dependency required for any packages on the root level?

File: packages/frontend/src/components/map/Map.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+function Map() {
+  // init variables for map
+  const attribute =
+    '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors';
+  const url = 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
+  const [userPosition, setUserPosition] = useState<LatLngTuple | null>(null);
+
+  // Get user position
+  const getPosition = () => {
```
Comment (CeeJayyy007) [inline packages/frontend/src/components/map/Map.tsx:16]: **nitpick (non-blocking):** We will fix issues like this to avoid lint warnings in the future

**Review by CeeJayyy007 — APPROVED** [review]

**Pull request merged**